src/traces/sankey/render.js

```diff
@@ -489,33 +487,42 @@ module.exports = function(svg, styledData, layout, callbacks) {
         .style('fill', 'none');
 
     var sankeyLink = sankeyLinks.selectAll('.' + c.cn.sankeyLink)
-        .data(function(d) {
-            var uniqueKeys = {};
```

Comment:
Why is the `uniqueKeys` stash not used anymore?

Reply:
It doesn't seem like we need a stash to make sure we generate unique keys for our links and nodes. See comment https://github.com/plotly/plotly.js/pull/3406/files#r249568603

Therefore I :hocho: it to simplify the code.